indico/modules/events/registration/api.py

```diff
@@ -38,34 +39,18 @@ class RHAPIRegistrant(RH):
 
     @oauth.require_oauth('registrants')
     def _checkParams(self):
-        event = ConferenceHolder().getById(request.view_args['event_id'], True)
-        if event is None:
+        self._event = ConferenceHolder().getById(request.view_args['event_id'], True)
+        if self._event is None:
             raise NotFound("No such event")
-        if not event.canManageRegistration(request.oauth.user):
+        if not self._event.canManageRegistration(request.oauth.user):
             raise Forbidden()
-        registrant = event.getRegistrantById(request.view_args['registrant_id'])
-        if registrant is None:
-            raise NotFound("No such registrant")
-        self.event = event
-        self.registrant = registrant
-
-    def _get_result(self, **kwargs):
-        checkin_date = None
-        if self.registrant.isCheckedIn():
-            checkin_date = self.registrant.getCheckInDate().isoformat()
-        return jsonify(event_id=self.event.getId(),
-                       registrant_id=self.registrant.getId(),
-                       full_name=self.registrant.getFullName(title=True, firstNameFirst=True),
-                       checked_in=self.registrant.isCheckedIn(),
-                       checkin_secret=self.registrant.getCheckInUUID(),
-                       checkin_date=checkin_date,
-                       registration_date=self.registrant.getRegistrationDate().isoformat(),
-                       **kwargs)
+
+        self._registration = Registration.find_one(id=request.view_args['registrant_id'])
```

Comment:
Also, use `self.event.registrations.filter_by(id=...).first_or_404()` to prevent people from accessing a registration of another event.